packages/manager/src/managers/sliceTemplateLibrary/SliceTemplateLibraryManager.ts

```diff
@@ -0,0 +1,179 @@
+};
+
+type SliceTemplateLibraryManagerCreateReturnType = {
+	data?: {
+		sliceIds: string[];
+	};
+	errors: (DecodeError | HookError)[];
+};
+
+type SliceTemplateLibraryReadData = {
+	templateIds?: string[];
+};
+
+type SliceTemplateLibraryCreateData = SliceTemplateLibraryReadData;
+
+const readHookCodec = t.type({
+	templates: t.array(
+		t.type({
+			model: SharedSlice,
+			componentContents: t.string,
+			mocks: t.array(SharedSliceContent),
+			screenshots: t.record(t.string, t.any),
+		}),
+	),
+});
+
+export class SliceTemplateLibraryManager extends BaseManager {
+	async read(
+		args: SliceTemplateLibraryReadData,
+	): Promise<SliceTemplateLibraryManagerReadReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		const isTypeScriptProject = await this.project.checkIsTypeScript();
+		const hookResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-template-library:read",
+			{ ...args, isTypeScriptProject },
+		);
+
+		const { data, errors } = decodeHookResult(readHookCodec, hookResult);
+
+		return {
+			templates: data.flat()[0].templates,
+			errors,
+		};
+	}
+	async create(
+		args: SliceTemplateLibraryCreateData,
+	): Promise<SliceTemplateLibraryManagerCreateReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		// Reading all available slice templates
+		const isTypeScriptProject = await this.project.checkIsTypeScript();
+		const hookReadResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-template-library:read",
+			{ ...args, isTypeScriptProject },
+		);
+		const { data: readData, errors: readErrors } = decodeHookResult(
+			readHookCodec,
+			hookReadResult,
+		);
+		if (readErrors.length > 0) {
+			return {
+				errors: readErrors,
+			};
+		}
+
+		// Extract all existing slice IDs into an array
+		const { libraries } = await this.slices.readAllSliceLibraries();
+		const existingIds: string[] = [];
+		libraries.forEach((lib) => {
+			if (lib.sliceIDs) {
+				existingIds.push(...lib.sliceIDs);
+			}
+		});
+
+		// Create a function to get the next available ID based on the baseId
+		const getNextAvailableSlice = (baseId: string) => {
+			let nextId = baseId;
+			let counter = 1;
+			while (existingIds.includes(nextId)) {
+				counter++;
+				nextId = `${baseId}_${counter}`;
+			}
+			existingIds.push(nextId);
+
+			return { id: nextId, counter };
+		};
+
+		// Extract the slices to create from the template ids given in args
+		const slicesToCreate = readData.flat()[0].templates.map((template) => {
+			const { id, counter } = getNextAvailableSlice(template.model.id);
+
+			return {
+				...template,
+				model: {
+					...template.model,
+					id,
+					name:
+						counter > 1 ? template.model.name + counter : template.model.name,
+				},
+			};
+		});
+
+		// Initiate the slice creation process for all slices
+		const creationPromises = slicesToCreate.map((slice) => {
+			assertPluginsInitialized(this.sliceMachinePluginRunner);
```

Comment:
`this.slices.createSlice()` calls `assertPluginsInitialized` so it is not needed here.

`assertPluginsInitialized` is primarily used to narrow the type of `this.sliceMachinePluginRunner` when you need access it directly, which we don't do here.

Reply:
done